src/react/provider.js

```diff
@@ -1,17 +1,23 @@
-import React, { useContext, useMemo, forwardRef } from 'react';
+import React, { useContext, useMemo, forwardRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 
 import { diRegistry } from './constants';
 import { Context } from './context';
 import { addInjectableToMap, getDisplayName, findInjectable } from './utils';
+import { globalDi } from './global';
 
-export const DiProvider = ({ children, use, target }) => {
+export const DiProvider = ({ children, use, target, global }) => {
   const { getDependencies } = useContext(Context);
 
   // memo provider value so gets computed only once
   const value = useMemo(() => {
     // create a map of dependency real -> replacements for fast lookup
-    const replacementMap = use.reduce(addInjectableToMap, new Map());
+    const replacementMap = use.reduce((acc, inj) => {
+      addInjectableToMap(acc, inj);
+      return acc;
+    }, new Map());
+    // supports global di if needed
+    globalDi._fromProvider(use, { global });
```

Comment:
wondering how this "use-memo-time" side effect is strict mode compatible.
Not sure if it will be duplicated here
https://github.com/albertogasparin/react-magnetic-di/blob/cc434831adbf74e419740012c240d7cf296a389f/src/react/utils.js#L23
With only one copy being removed here https://github.com/albertogasparin/react-magnetic-di/pull/85/files#diff-4924b64beae31869db09ad5bdf2343784be3180c401d09c3204f53676cf2cb2fR40

Reply:
Moved to set so this is no longer a problem (Set will dedupe by default)